lib/iris/analysis/__init__.py

```diff
@@ -2510,17 +2545,19 @@ def _compute_shared_coords(self):
                     raise ValueError(msg)
 
             try:
-                self.coords.append(
-                    coord.copy(points=new_points, bounds=new_bounds)
-                )
+                new_coord = coord.copy(points=new_points, bounds=new_bounds)
             except ValueError:
                 # non monotonic points/bounds
-                self.coords.append(
-                    iris.coords.AuxCoord.from_coord(coord).copy(
-                        points=new_points, bounds=new_bounds
-                    )
+                new_coord = iris.coords.AuxCoord.from_coord(coord).copy(
+                    points=new_points, bounds=new_bounds
                 )
 
+            if climatological_coord:
+                new_coord.climatological = True
+
+            self.coords.append(new_coord)
+            self.coord_replacement_mapping.append((coord, new_coord))
```

Comment:
Maybe make `coord_replacement_mapping` a dictionary, then
```suggestion
            self.coord_replacement_mapping[new_coord] = coord
```